File: typescript/P0002/sum_even_valued_fibo/test/Solution.spec.ts
import {Solution} from '../src/Solution';

describe('Helpers', () => {
  test('The 6th (0 based index) Fibonacci number" should "be 8', () => {
    const ans = Solution.fibonacci(6);
    expect(ans).toBe(8);
  });

  test('The sum of the even-valued terms of the Fibonacci sequence below 4000000 should be 4613732', () => {
    const ans = Solution.sumEvenValuedFiboBelow(4000000);
    expect(ans).toBe(4613732);
  });
});

File: oldProjectStructure/haskell/P0002/sum-even-valued-fibo/test/SolutionSpec.hs
module SolutionSpec where

import Solution

import Test.Hspec (Spec, describe, context, it, shouldBe)

-- |Required for auto-discpvery
spec :: Spec
spec =
  describe "Project Euler Problem2 Solution" $ do

    describe "fibonacci 6" $ do
      it "calculates the 6th (0 index based) Fibonacci's number" $ do
        show (fibonacci 6) `shouldBe` "8"

    describe "sumEvenFibonacciUntil 4000000" $ do
      it "calculates the sum of the even-valued terms of Fibonacci below 4000000" $ do
        show (sumEvenFibonacciUntil 4000000) `shouldBe` "4613732"

